curriculum/challenges/english/25-front-end-development/workshop-email-simulator/6853f4ded9f84896ff182dd8.md

```diff
@@ -0,0 +1,120 @@
+---
+id: 6853f4ded9f84896ff182dd8
+title: Step 19
+challengeType: 20
+dashedName: step-19
+---
+
+# --description--
+
+After the empty inbox check, add a print statement with the message `'\nYour Emails:'`. After that, iterate over all emails in the inbox using a `for` loop. Use enumeration to number the emails starting from `1`. Unpack the enumeration into `i` (the index) and `email` (the email object). Inside the loop, print each email using the `__str__` method you defined earlier, formatted with its number like this:
+
+```md
+1. [Formatted email summary from __str__ method]
+2. [Formatted email summary from __str__ method]
+.
+.
+.
+n. [Formatted email summary from __str__ method]
+```
+
+# --hints--
+
+You should print the header `'\nYour Emails:'` outside the `if` statement.
+
+```js
+({ 
+    test: () => 
+    {
+        assert(runPython(`_Node(_code).find_class("Inbox").find_function("list_emails").find_body()[1].is_equivalent("print('\\\\nYour Emails:')")`))
+    }
+})
+```
+
+The enumeration should unpack into `i` and `email`.
+
+```js
+({
+    test: () => 
+    {
+        assert(runPython(`_Node(_code).find_class("Inbox").find_function("list_emails").find_for_loops()[0].find_for_vars().is_equivalent("i,email")`))
+    }
+})
+```
+
+The enumeration should take `self.emails` and `start=1` as arguments.
+
+```js
+({
+    test: () => 
+    {
+        assert(runPython(`_Node(_code).find_class("Inbox").find_function("list_emails").find_for_loops()[0].find_for_iter().is_equivalent("enumerate(self.emails, start=1)")`))
+    }
+})
+```
+
+Inside the loop, you should print each email summary with a numbered prefix followed by the formatted string from the `__str__` method. Don't forget to add a `.` after the number.
+
+```js
+({
+    test: () => 
+    {
+        assert(runPython(`_Node(_code).find_class("Inbox").find_function("list_emails").find_for_loops()[0].find_body().is_equivalent("print(f'{i}. {str(email)}')")`))
```

Comment:
Actually you don't need to use `str(email)` within the f-string. `print(f'{i}. {email}')` does the same. So we should remove it from the solution but allow both

Reply:
@zairahira

Was this change applied?